### ISSUE TITLE
Mark aliases

### ISSUE THREAD

**Issue by BioPhoton** [ReactiveX/rxjs#4582]
### Please check those that apply

- [ ] typo
- [X] documentation feature suggestion
- [ ] documentation needs clarification
- [ ] error(s) in example
- [ ] needs example

### Description Of The Issue

It would be nice if the alias operators could be marked like "(alias flatMap)" or so.

**Comment by kwonoj** [comment]
I agree with this.

**Comment by BioPhoton** [comment]
I would also not expose the aliases as separate operators in the rxjs.dev/api docs.

**Comment by JWO719** [comment]
Definetly it would be cool! I just fear it is not that easy.

So the problem is that we are refering to the same exported member see 
```typescript
export { mergeMap } from '../internal/operators/mergeMap';
export { mergeMap as flatMap } from '../internal/operators/mergeMap';
```
Therefore the documentation inside flatMap also refers to mergeMap all the time. 
Currently we have no mechanism inside the docs to detect such "duplicates", also 
we would need some way to distinguish between the "original" one and the "alias".
For now I can't think of a solution for these problems.

I hope I could somehow express my concerns so that they are understandable (not really sure :D )

**Comment by BioPhoton** [comment]
Maybe the `@nodoc` tag helps...

**Comment by santoshyadav198613** [comment]
Hi @BioPhoton ,
Is this still open, let me know if i can be of any help here.

**Comment by BioPhoton** [comment]
Hi!
I guess @JWO719 knows best how to fix it. I have no clue.

Do you have any idea?

**Comment by JWO719** [comment]
I still don't know a proper solution for this problem. If anyone has an idea how to mark aliases or highlight it in a better way, I'm open for any suggestions and happy to review a pr!

**Comment by santoshyadav198613** [comment]
Ok let me check if i can find any solution.

**Comment by martinsik** [comment]
@JWO719 This repo already has `typescript` as a dev dependency so there could be another dgeni processor that only checks public entry points and from `export` statements figures out what is an alias (this would work as `tslint` extension as well).

```
import * as fs from 'fs';
import { createSourceFile, SyntaxKind, ScriptTarget, ExportDeclaration, Statement } from 'typescript';

const filename = 'src/internal/operators/index.ts';
const sourceFile = createSourceFile(filename, fs.readFileSync(filename).toString(), ScriptTarget.ES5);

const isExportDeclaration = (s: Statement): s is ExportDeclaration => s.kind === SyntaxKind.ExportDeclaration;

sourceFile.statements.forEach(statement => {
  if (isExportDeclaration(statement) && statement.exportClause) {
    statement.exportClause.elements.forEach(element => {
      if (element.propertyName && element.name) {
        const exportAlias = element.name.text;
        const exportName = element.propertyName.text;

        console.log(exportName, '=>', exportAlias);
      }
    });
  }
});

// ...
```

This is expecting that all aliases are going to be just renamed exports that are already exported (like `flatMap` and `mergeMap`).  I think it's a question whether also exports like these should be considered as alias:

```
export { asap as asapScheduler } from './internal/scheduler/asap';
```

**Comment by JWO719** [comment]
closed by #4733
